multiplex.go

```diff
@@ -62,6 +64,13 @@ type Multiplex struct {
 	chLock   sync.Mutex
 }
 
+func init() {
+	// this context is here to allow us to attach deadlines to messages without
+	// spawning a new goroutine for propagateCancel each time.
+	// we ignore the cancel function because it will never get canceled.
+	bgCtx, _ = context.WithCancel(context.Background())
```

Comment:
Do we actually need to create a new context here? Why aren't we able to use `context.Background()` directly?

Reply:
I think it will have `propagateCancel` goroutines because it's not hooked to a cancellation.

Reply:
It shouldn't, as this guard would short circuit the propagation: https://github.com/golang/go/blob/master/src/context/context.go#L245

Reply:
ok, will get rid of it then.

Reply:
done